packages/bundle-source/src/types.js

```diff
@@ -3,23 +3,60 @@
  */
 
 /**
- * @callback BundleSource
+ * @typedef {BundleSourceEndoZipBase64 & BundleSourceGetExport & BundleSourceNestedEvaluate} BundleSource
```

Comment:
I’m surprised that this is intersection instead of union but this is clearly where the money is made.

Reply:
Intersection means the result has characteristics of all the types.  Union means it might be any of the types (and you need to sniff it at runtime or cast it to determine which one you actually have).